src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,89 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] buckets;
+    private float threshold;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            growth();
+        }
+        int index = getBucketNumber(key, buckets.length);
+        Node<K, V> node = findNode(key, index);
+        if (node != null) {
+            node.value = value;
+        } else {
+            Node<K, V> newBucketNode = new Node<>(key, value, buckets[index]);
+            buckets[index] = newBucketNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = findNode(key, getBucketNumber(key, buckets.length));
+        if (node != null) {
+            return node.value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node findNode(K key, int index) {
+        Node<K, V> node = buckets[index];
+        while (node != null) {
+            if (key == node.key || key != null && key.equals(node.key)) {
+                return node;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    private void growth() {
+        Node<K, V>[] newBuckets = new Node[buckets.length * 2];
+        threshold = newBuckets.length * DEFAULT_LOAD_FACTOR;
+        size = 0;
+        Node<K, V>[] oldBuckets = buckets;
+        buckets = newBuckets;
+        for (Node<K, V> bucket : oldBuckets) {
+            while (bucket != null) {
+                put(bucket.key, bucket.value);
+                bucket = bucket.next;
+            }
+        }
+    }
+
+    private int getBucketNumber(K key, int length) {
```

Comment:
```suggestion
    private int getBucketNumber(K key) {
```
try to solve without `length`

Reply:
changed method implementation, used only key.